pkg/trie/triedb/triedb.go

```diff
@@ -267,6 +342,158 @@ func (t *TrieDB) inspect(
 	}
 }
 
+// fix is a helper function to reorganise the nodes after deleting a branch.
+// For example, if the node we are deleting is the only child for a branch node, we can transform that branch in a leaf
+func (t *TrieDB) fix(node Node) (Node, error) {
+	usedIndex := make([]byte, 0)
+
+	switch branch := node.(type) {
+	case Branch:
+		for i := 0; i < codec.ChildrenCapacity; i++ {
+			if branch.children[i] != nil {
+				if len(usedIndex) == 2 {
+					break
+				}
+				usedIndex = append(usedIndex, byte(i))
+			}
+		}
+
+		if len(usedIndex) == 0 {
+			if branch.value == nil {
```

Comment:
Fure sure that's not only possible but actually is something that happens in real life very often

Reply:
So, in a case where we are deleting a branch child with no value, we should not panic, right?

Reply:
Let me check that case bc [parity is doing the same check](https://github.com/paritytech/trie/blob/c3348e4b335443c17586a8b7491ba893cfde76a6/trie-db/src/triedbmut.rs#L1577). Maybe I'm missing something

Reply:
I've been checking and this is just a check to consider that case but should never happen, since [we are doing nothing with branches without values](https://github.com/ChainSafe/gossamer/blob/1a26ecc9bf07550b3bb207e6eef4691a83ff61fa/pkg/trie/triedb/triedb.go#L427-L439)

Reply:
Okay, that makes sense!